### PR TITLE
Modify EI Analytics feature

### PR DESCRIPTION
**Pull request by isharailanga**
## Purpose
Modify the EI_Analytics feature

### CODE REVIEW

File: components/ei-analytics/org.wso2.analytics.solutions.ei.analytics/widgets/EIAnalyticsMessageTable/src/EIAnalyticsMessageTable.jsx [hunk 1]
```diff
@@ -121,8 +121,15 @@ class EIAnalyticsMessageTable extends Widget {
         super.subscribe(this.handlePublisherParameters);
     }
 
-    handlePublisherParameters(message) {
-        if ('granularity' in message) {
+    handlePublisherParameters(recievedMessage) {
+        let message;
+        if (typeof recievedMessage === "string") {
+            message = JSON.parse(recievedMessage);
+        }else {
```
Comment (this) [inline components/ei-analytics/org.wso2.analytics.solutions.ei.analytics/widgets/EIAnalyticsMessageTable/src/EIAnalyticsMessageTable.jsx]: Code formatting

File: components/ei-analytics/org.wso2.analytics.solutions.ei.analytics/widgets/EIAnalyticsMessageTable/src/EIAnalyticsMessageTable.jsx [hunk 2]
```diff
@@ -131,6 +138,12 @@ class EIAnalyticsMessageTable extends Widget {
                 data: []
             }, this.handleGraphUpdate);
         }
+
+       if (message.selectedComponent) {
+        this.setState({
+            componentName: message.selectedComponent
+        }, this.handleGraphUpdate);
```
Comment (this) [inline components/ei-analytics/org.wso2.analytics.solutions.ei.analytics/widgets/EIAnalyticsMessageTable/src/EIAnalyticsMessageTable.jsx]: Why pass 2 parameters to the `setState()` function? Is that your intention?

File: components/ei-analytics/org.wso2.analytics.solutions.ei.analytics/widgets/EIAnalyticsMessageTable/src/EIAnalyticsMessageTable.jsx [hunk 3]
```diff
@@ -121,8 +121,15 @@ class EIAnalyticsMessageTable extends Widget {
         super.subscribe(this.handlePublisherParameters);
     }
 
-    handlePublisherParameters(message) {
-        if ('granularity' in message) {
+    handlePublisherParameters(recievedMessage) {
+        let message;
```
Comment (this) [inline components/ei-analytics/org.wso2.analytics.solutions.ei.analytics/widgets/EIAnalyticsMessageTable/src/EIAnalyticsMessageTable.jsx]: Lets reduce the following block into a one line
`const message = (typeof recievedMessage === 'string') ? JSON.parse( ) : recievedMessage;`

Comment (isharailanga) [reply components/ei-analytics/org.wso2.analytics.solutions.ei.analytics/widgets/EIAnalyticsMessageTable/src/EIAnalyticsMessageTable.jsx]: Fixed at 232cc7dff2df7a50828020f9fca1dac61295ef44

**Pull request merged**